buildSrc/subprojects/performance/src/main/groovy/org/gradle/testing/DistributedPerformanceTest.groovy

```diff
@@ -338,35 +339,35 @@ class DistributedPerformanceTest extends PerformanceTest {
                     it.@name.text() == zipName
                 }
                 if (fileNode) {
-                    def resultsDirectory = new File(workerTestResultsTempDir, jobId)
                     def contentUri = fileNode.content.@href.text()
                     client.get(path: contentUri, contentType: ContentType.BINARY) {
                         resp, inputStream ->
-                            unzippedFiles = unzipToDirectory(inputStream, resultsDirectory)
+                            junitTestSuites = parseXmlsInZip(inputStream)
                     }
                 }
             }
         }
-        unzippedFiles
+        junitTestSuites
     }
 
-    @TypeChecked(TypeCheckingMode.SKIP)
-    def unzipToDirectory(inputStream, destination) {
-        def unzippedFiles = []
+    List<JUnitTestSuite> parseXmlsInZip(InputStream inputStream) {
+        List<JUnitTestSuite> parsedXmls = []
         new ZipInputStream(inputStream).withStream { zipInput ->
             def entry
             while (entry = zipInput.nextEntry) {
-                if (!entry.isDirectory()) {
-                    def file = new File(destination, entry.name)
-                    file.parentFile?.mkdirs()
-                    new FileOutputStream(file).withStream {
-                        it << zipInput
+                if (!entry.isDirectory() && entry.name.endsWith('.xml')) {
+                    PipedOutputStream os = new PipedOutputStream()
+                    PipedInputStream is = new PipedInputStream(os)
+                    Thread.start {
+                        os.withStream {
+                            it << zipInput
+                        }
                     }
-                    unzippedFiles << file
+                    parsedXmls.add(JUnitMarshalling.unmarshalTestSuite(is))
```

Comment:
Creating the JAXBContext is expensive and JAXBContext is thread safe. So the thing `JUnitMarshalling` is doing is not a good idea: https://github.com/Open-MBEE/junit-xml-parser/blob/master/src/main/java/org/openmbee/junit/JUnitMarshalling.java#L23

Maybe they are open to PRs, so we could have a `JUnitMarshaller` which can unmarshall test suites which creates the JAXBContext only once.

It is not a big deal for us I guess...

Reply:
I guess so...